qiskit/pulse/schedule.py

```diff
@@ -56,10 +57,11 @@ def __init__(self, *schedules: List[Union[ScheduleComponent, Tuple[int, Schedule
                 sched_timeslots = sched.timeslots
                 if insert_time:
                     sched_timeslots = sched_timeslots.shift(insert_time)
-                timeslots.append(sched_timeslots.timeslots)
+                if not self._timeslots.is_mergeable_with(sched_timeslots):
```

Comment:
Perhaps we should offload this check too merging of timeslots, ie., enforce that no `TimeslotCollection` may be instantiated that overlap? We could then capture an overlapping timeslot with a try? Thoughts?

Reply:
That's what I too considered and have no clear answer. My current answer is "merged() without overlap check would be useful for future". I'd like to have a way to internally skip overlap check for the cases that no overlap is assured (e.g. `Schedule.append`) for the future. (I've seen one of the next bottlenecks is in overlap check) I'm assuming users of TimeslotCollection are developers of schedule modules, who can know when they do (or do not) have to check overlap. To declare that, I wish I would use `_timeslots.py` but it's not allowed in qiskit convention now, right?

Reply:
FYI: Call tree after d0fedf1
![call_tree_d0fedf1](https://user-images.githubusercontent.com/15028342/61844075-239ae200-aed9-11e9-8f33-e4143346cb7d.png)